lib/trip_dispatcher.rb

```diff
@@ -33,44 +37,106 @@ def load_users(filename)
     def load_trips(filename)
       trips = []
       trip_data = CSV.open(filename, 'r', headers: true,
-                                          header_converters: :symbol)
+        header_converters: :symbol)
+
+        trip_data.each do |raw_trip|
+          passenger = find_passenger(raw_trip[:passenger_id].to_i)
+          driver = find_driver(raw_trip[:driver_id].to_i)
+
+          parsed_trip = {
+            id: raw_trip[:id].to_i,
+            passenger: passenger,
+            #2018-06-07 04:18:47 -0700
+            start_time: Time.parse(raw_trip[:start_time]),
+            #,2018-06-07 04:19:25 -0700
+            end_time: Time.parse(raw_trip[:end_time]),
+            cost: raw_trip[:cost].to_f,
+            rating: raw_trip[:rating].to_i,
+            driver: driver
+          }
+
+          create_trip(parsed_trip, driver, passenger, trips)
+        end
+
+        return trips
+      end
 
-      trip_data.each do |raw_trip|
-        passenger = find_passenger(raw_trip[:passenger_id].to_i)
 
-        parsed_trip = {
-          id: raw_trip[:id].to_i,
-          passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
-          cost: raw_trip[:cost].to_f,
-          rating: raw_trip[:rating].to_i
-        }
+      def load_drivers(filename)
+        drivers = []
+        driver_data = CSV.open(filename, 'r', headers: true, header_converters: :symbol)
+
+        driver_data.each do |raw_driver|
+          user_with_same_id = find_passenger(raw_driver[0].to_i)
+
+          parsed_driver = {
+            id: raw_driver[0].to_i,
+            vin: raw_driver[1],
+            name: user_with_same_id.name,
+            phone_number: user_with_same_id.phone_number,
+            status: raw_driver[2].to_sym
+          }
+
+          driver = Driver.new(parsed_driver)
+          drivers << driver
+        end
+
+        return drivers
+      end
+
+
+      def find_passenger(id)
+        check_id(id)
+        return @passengers.find { |passenger| passenger.id == id }
+      end
+
+
+      def find_driver(id)
+        check_id(id)
+        return @drivers.find { |driver| driver.id == id }
+      end
 
-        trip = Trip.new(parsed_trip)
+      def create_trip(trip_data, driver, passenger, trip_list)
+        trip = Trip.new(trip_data)
         passenger.add_trip(trip)
-        trips << trip
+        driver.add_driven_trip(trip)
+        trip_list << trip
+        return trip
       end
 
-      return trips
-    end
 
-    def find_passenger(id)
-      check_id(id)
-      return @passengers.find { |passenger| passenger.id == id }
-    end
+      def request_trip(user_id)
+        available_drivers = @drivers.select { |driver| driver.status == :AVAILABLE}
```

Comment:
This line would be a good candidate for being moved into its own "helper" method. Doing so would be in keeping with the "separation of concerns" concept discussed in POODR.